sql/src/main/java/io/crate/exceptions/SQLExceptions.java

```diff
@@ -114,20 +112,14 @@ public static boolean isShardFailure(Throwable e) {
      * If concrete {@link ElasticsearchException} is found, first transform it
      * to a {@link CrateException}
      */
-    public static SQLActionException createSQLActionException(Throwable e, ExceptionAuthorizedValidator validator) {
+    public static SQLActionException createSQLActionException(Throwable e) {
         // ideally this method would be a static factory method in SQLActionException,
         // but that would pull too many dependencies for the client
 
         if (e instanceof SQLActionException) {
             return (SQLActionException) e;
         }
         e = esToCrateException(e);
-        try {
-            validator.ensureExceptionAuthorized(e);
-        } catch (MissingPrivilegeException mpe) {
-            e = mpe;
-        }
```

Comment:
Shouldn't the validator ensure we never see exceptions if we don't have access to the object at all?